apps/dashboard/app/javascript/module_browser.js

```diff
@@ -0,0 +1,23 @@
+document.addEventListener('DOMContentLoaded', function () {
+  const moduleSearch = document.getElementById('module_search');
+  const clusterFilter = document.getElementById('cluster_filter');
+
+  function filterModules() {
+    const searchQuery = moduleSearch.value.toLowerCase();
+    const selectedCluster = clusterFilter.value;
+
+    const modules = document.querySelectorAll('[data-name][data-clusters]');
+    modules.forEach(function (module) {
+      const name = module.getAttribute('data-name').toLowerCase();
+      const clusters = module.getAttribute('data-clusters').split(',');
+
+      const searchMatches = name.includes(searchQuery);
+      const clusterMatches = !selectedCluster || clusters.includes(selectedCluster);
+
+      module.style.display = (searchMatches && clusterMatches) ? '' : 'none';
```

Comment:
This is setting an inline style - can we use the the `hide` and `show` functions from utils? I think they query by ID though - so maybe we need to enhance that to accept an `HTMLElement` as well.